docs/sources/setup-grafana/configure-security/configure-authentication/okta/index.md

```diff
@@ -1,166 +1,128 @@
 
-If the application role received by Grafana is `GrafanaAdmin`, Grafana grants the user server administrator privileges.
-This is useful if you want to grant server administrator privileges to a subset of users.
-Grafana also assigns the user the `Admin` role of the default organization.
+1. To enable the `Refresh Token`, grant type in the `General Settings` section of the Okta application.
+1. Extend the `scopes` in `[auth.okta]` with `offline_access`.
 
-The setting `allow_assign_grafana_admin` under `[auth.okta]` must be set to `true` for this to work.
-If the setting is set to `false`, the user is assigned the role of `Admin` of the default organization, but not server administrator privileges.
+### Configure role mapping
 
-```ini
-allow_assign_grafana_admin = true
-```
+> **Note:** Unless `skip_org_role_sync` option is enabled, the user's role will be set to the role retrieved from the auth provider upon user login.
 
-Example:
+The user's role is retrieved using a [JMESPath](http://jmespath.org/examples.html) expression from the `role_attribute_path` configuration option against the `api_url` endpoint payload.
+To map the server administrator role, use the `allow_assign_grafana_admin` configuration option.
+Refer to [configuration options]({{< relref "../generic-oauth/index.md#configuration-options" >}}) for more information.
 
-```ini
-role_attribute_path = contains(groups[*], 'admin') && 'GrafanaAdmin' || contains(groups[*], 'editor') && 'Editor' || 'Viewer'
-```
+If no valid role is found, the user is assigned the role specified by [the `auto_assign_org_role` option]({{< relref "../../../configure-grafana#auto_assign_org_role" >}}).
+You can disable this default role assignment by setting `role_attribute_strict = true`.
+This setting denies user access if no role or an invalid role is returned.
 
-## Skip organization role sync
+Read about how to [add custom claims](https://developer.okta.com/docs/guides/customize-tokens-returned-from-okta/main/#add-a-custom-claim-to-a-token) to the user info in Okta. Also, check Generic OAuth page for [JMESPath examples]({{< relref "../generic-oauth/index.md#role-mapping-examples" >}}).
 
-To prevent the sync of org roles from Okta, set `skip_org_role_sync` to `true`. This is useful if you want to manage the organization roles for your users from within Grafana.
+### Configure team synchronization (Enterprise only)
 
-```ini
-[auth.okta]
-# ..
-# prevents the sync of org roles from Okta
-skip_org_role_sync = true
-```
+> **Note:** Available in [Grafana Enterprise]({{< relref "../../../../introduction/grafana-enterprise" >}}) and [Grafana Cloud]({{< relref "../../../../introduction/grafana-cloud" >}}).
 
-### Team Sync (Enterprise only)
+By using Team Sync, you can link your OAuth2 groups to teams within Grafana. This will automatically assign users to the appropriate teams.
 
 Map your Okta groups to teams in Grafana so that your users will automatically be added to
```

Comment:
This text here basically repeats what in Line 121. Can this line be removed? And maybe change Line 121 to "By using Team Sync, you can link your Okta OAuth2 groups..."?